tests/unit/lms/services/blackboard_api/client_test.py

```diff
@@ -0,0 +1,162 @@
+            sentinel.file_2,
+            sentinel.file_3,
+        ]
+
+        files = svc.list_files("COURSE_ID")
+
+        basic_blackboard_api_client.request.assert_called_once_with(
+            "GET", "courses/uuid:COURSE_ID/resources?type=file&limit=200"
+        )
+        BlackboardListFilesSchema.assert_called_once_with(
+            basic_blackboard_api_client.request.return_value
+        )
+        assert files == blackboard_list_files_schema.parse.return_value
+
+    def test_it_with_pagination(
+        self, svc, basic_blackboard_api_client, blackboard_list_files_schema
+    ):
+        # Each response from the Blackboard API includes the path to the next
+        # page in the JSON body. This is the whole path to the next page,
+        # including limit and offset query params, as a string. For example:
+        # "/learn/api/public/v1/courses/uuid:<ID>/resources?limit=200&offset=200"
+        #
+        basic_blackboard_api_client.request.side_effect = [
+            factories.requests.Response(
+                json_data={"paging": {"nextPage": "PAGE_2_PATH"}}
+            ),
+            factories.requests.Response(
+                json_data={"paging": {"nextPage": "PAGE_3_PATH"}}
+            ),
+            factories.requests.Response(json_data={}),
+        ]
+
+        # Each Blackboard API response contains a page of results.
+        blackboard_list_files_schema.parse.side_effect = [
+            [sentinel.file_1, sentinel.file_2, sentinel.file_3],
+            [sentinel.file_4, sentinel.file_5, sentinel.file_6],
+            [sentinel.file_7, sentinel.file_8],
+        ]
+
+        files = svc.list_files("COURSE_ID")
+
+        # It called the Blackboard API three times getting the three pages.
+        assert basic_blackboard_api_client.request.call_args_list == [
+            call("GET", "courses/uuid:COURSE_ID/resources?type=file&limit=200"),
+            call("GET", "PAGE_2_PATH"),
+            call("GET", "PAGE_3_PATH"),
+        ]
+        # It returned all three pages of files as a single list.
+        assert files == [
+            sentinel.file_1,
+            sentinel.file_2,
+            sentinel.file_3,
+            sentinel.file_4,
+            sentinel.file_5,
+            sentinel.file_6,
+            sentinel.file_7,
+            sentinel.file_8,
+        ]
+
+    def test_it_doesnt_send_paginated_requests_forever(
+        self, svc, basic_blackboard_api_client, blackboard_list_files_schema
+    ):
+        # Make the Blackboard API send next page paths forever.
+        basic_blackboard_api_client.request.return_value = factories.requests.Response(
+            json_data={"paging": {"nextPage": "NEXT_PAGE"}}
+        )
+
+        files = svc.list_files("COURSE_ID")
+
+        assert basic_blackboard_api_client.request.call_count == PAGINATION_MAX_REQUESTS
+        assert len(files) == PAGINATION_MAX_REQUESTS * len(
+            blackboard_list_files_schema.parse.return_value
+        )
```

Comment:
These are the same exact tests that used to live in the view, now moved into the service